sample_metadata/parser/generic_parser.py

```diff
@@ -66,7 +67,8 @@
 )
 
 rmatch = re.compile(r'[_\.-][Rr]\d')
-GroupedRow = Union[List[Dict[str, Any]], Dict[str, Any]]
+SingleRow = Dict[str, Any]
+GroupedRow = Union[List[SingleRow], SingleRow]
```

Comment:
Could we refactor this union out, as this script has got more complex, it just makes us double handle list checks - I think the parser should just _always_ make sure it's a list. 

```suggestion
GroupedRow = List[SingleRow]
```